html/beps/bep_encrypted_data.rst

```diff
@@ -0,0 +1,297 @@
+Similarly Message Stream Encryption provides limited protection from passive eavesdroppers on the network layer but does not prevent the infohash from escaping.   
+
+
+Instead of attempting to restrict access to the swarm or metadata this BEP proposes to make all data opaque to 3rd parties by encrypting it with a shared secret that is not available through any torrent-related protocol, i.e. must be obtained separately by the user.
+
+In principle the same properties can be provided by simply storing the data in an encrypted archive and using a nondescript filename, but that requires users to store the data twice or to use additional filesystem layers to transparently access the data, which is even more cumbersome when encryption is involved. It also prevents bittorrent clients from reusing already-downloaded files in a multi-file torrent. 
+
+Metadata format
+===============
+
+
+.. parsed-literal::
+
+    {
+      info: {
+        bepXX: {
+          mac: *<32bytes of hmac output (string)>*,
+          salt: *<32bytes of random binary data (string)>*,
+          shadow: *<optional, bencoded-then-encrypted dictionary (string)>*
+          v: *<version (integer)>*,
+        },
+        length *or* files: *<unchanged>*,
+        name: *<public name (string)>*,
+        piece length: *<unchanged>*
+        pieces: *<N*20 bytes, piece hashes of the payload ciphertext (string)>*
+      },
+    }
+
+
+``salt``
+  the random data must be generated by a cryptographically secure RNG to avoid IV reuse.
+
+``v``
+  The protocol version used to encrypt the torrent, currently *1*. New versions may be introduced by updates to this BEP if cryptographic weaknesses necessitate incompatible changes. Implementations should check if they support the version indicated in the metadata file and otherwise inform the user that they can download the data but not decrypt it.
+
+``shadow``
+  bencoded-then-encrypted dictionary of key-value pairs that shadow entries in the info dictionary.
+  If it is absent only the payload is encrypted and all info dictionary entries are non-shadowed.
+  Implementations should only shadow a whitelist of keys which they have a shadowing strategy and ignore other keys.
+  Shadowable keys suggested by this BEP: ``length, files, name, comment``.
+
+``mac``
+  message authentication code covering the info dictionary
+
+``name``
+  the name field is a mandatory part of [BEP 3]_, therefore a placeholder MUST be provided if a shadow name is used. An implementation may either generate a random string consisting of filesystem-friendly characters or allow the user to choose a public name that reveals less information than the shadow name.
```

Comment:
I think the first sentence could be simplified and made slightly less ambiguous. I assume you meant one of:

1. a name field is always necessary, to be backwards compatible
2. a name field is only necessary if there is *no* shadow key for it (but this would seem problematic)